### PR TITLE
Create `Backbone` base class

### PR DESCRIPTION
**Pull request by jbischof**
Pilot for how base classes could improve the maintainability of our library. The `Backbone` class can be easily extended to other model types and `Task`, `Preprocessor`, and `Tokenizer` base classes can also be added.

I've also simplified the `from_preset` docstring so we don't have to write a lot of custom copy for each model class. We already have all the information we need in the class docstring.

Part of #530

### CODE REVIEW

File: keras_nlp/models/base.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        return {}
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        load_weights=True,
+        **kwargs,
+    ):
+        """Instantiate {model_name} model from preset architecture and weights.
```
Comment (mattdangerw) [inline keras_nlp/models/base.py:44]: I would just do the class name everywhere in this format string. Keeps things simpler and more automated. And when we refer to the model we have done that stylized as `BERT` right? So probably better to stick to `BertBackbone` here.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Thanks! This will be a nice change. Left some comments.

File: keras_nlp/models/base.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        Args:
+            preset: string. Must be one of {preset_names}.
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `True`.
+
+        Examples:
+        ```python
+        # Load architecture and weights from preset
+        model = {model_name}Backbone.from_preset({example_preset_name})
```
[the remark this answers is not in the document]
Comment (jbischof) [reply keras_nlp/models/base.py:54]: I get this error: 
```
AttributeError: attribute '__doc__' of 'method' objects is not writable
```

I wonder why it works as a decorator and not posthoc...

[the remark this answers is not in the document]
Comment (jbischof) [reply keras_nlp/models/base.py:54]: Here's what seems to work:
```python
format_docstring(
    model_name=BertBackbone.__name__,
    example_preset_name="bert_base_en_uncased",
    preset_names=BertBackbone.presets,
)(BertBackbone.from_preset.__func__)
```

[the remark this answers is not in the document]
Comment (mattdangerw) [reply keras_nlp/models/base.py]: kinda funky, but I'm down to do this for now! and at least our "formattable docstring" style is consistent

we can always update if we find a better recipe

**Review by mattdangerw — APPROVED** [review]
LGTM! Thanks!

File: keras_nlp/models/backbone.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        Args:
+            preset: string. Must be one of {{preset_names}}.
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `True`.
+
+        Examples:
+        ```python
+        # Load architecture and weights from preset
+        model = {{model_name}}Backbone.from_preset({{example_preset_name}})
```
Comment (mattdangerw) [inline keras_nlp/models/backbone.py:54]: Oops missed a spot, this should be `{{model_name}}.from_preset` (remove the Backbone)

File: keras_nlp/models/backbone.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        # Load randomly initialized model from preset architecture
+        model = {{model_name}}Backbone.from_preset(
```
Comment (mattdangerw) [inline keras_nlp/models/backbone.py:57]: Same here

**Review by mattdangerw — APPROVED** [review]

**Pull request merged**